### PR TITLE
Cleanup constants

### LINKED ISSUE #34: Reverse order of the `V_I` and `S_I` constants
**Describe what you want implemented** 
Reverse the order of the constants `V_I` and `S_I`.

**Describe "Why" this is needed**
With the reverse order, the constants map more intuitively to their s_i and v_i counterparts of the paper.
E.g. currently:
s_1 = `S_I[26]` , s_2 = `S_I[25]`, ... , s_27 = `S_I[0]`
and after the change:
s_1 = `S_I[0]` , s_2 = `S_I[1]`, ... , s_27 = `S_I[26]`

**Describe alternatives you've considered**
We can leave as is since the values are not incorrect but the current non-intuitive storage is a breeding ground for future bugs.

**Additional context**
NA

### LINKED ISSUE #39: use `BlsScalar::zero()` and `BlsScalar::one()` instead of constants `BLS_ZERO` and `BLS_ONE`
**Describe what you want implemented** 
use `BlsScalar::zero()` and `BlsScalar::one()` instead of constants `BLS_ZERO` and `BLS_ONE`

**Describe "Why" this is needed**
Since we already have constant functions, there is no need to add the `BlsScalars` as constants as well.

**Describe alternatives you've considered**
n.a

**Additional context**
https://github.com/dusk-network/zelbet.next/pull/32#discussion_r814259329

### PR DESCRIPTION
**Pull request by moCello**
In this PR:
- Rename `DECOMPOSITION_MIN_ONE` to `V_I`
- Reverse order of `S_I` and `V_I`
- Add const functions for generating `S_I_BLS`, `DIVISOR`, `RECIP` and `S_BOX_BLS`
- Remove constants `BLS_ZERO` and `BLS_ONE` in favor of `const fn BlsScalar::zero()` and `const fn BlsScalar::one()` respectively
- Small fixes of comments and gitignore
- Add example for `zelbet` and `sponge` gadget

### CODE REVIEW

File: src/constants.rs [hunk 1]
```diff
@@ -7,81 +7,178 @@
...
-    0x0000000000000000,
-    0x0000000000000000,
-    0x0000000000000000,
-    0x0000000000000000,
-]);
-
-/// BlsScalar one
-pub const BLS_ONE: Scalar = Scalar([
+/// BlsScalar with value 1
+const BLS_ONE: BlsScalar = BlsScalar([
```
Comment (vlopes11) [inline src/constants.rs]: Its unclear why do we need a constant for [BlsScalar::one](https://docs.rs/bls12_381/0.6.0/bls12_381/struct.Scalar.html#method.one)

File: src/hash/zelbet/bar.rs [hunk 2]
```diff
@@ -4,47 +4,39 @@
...
-use crate::constants::{DIVISORS, P_PRIME, RECIP, S_BOX_U16, S_I_BLS};
+use crate::constants::{DIVISORS, P_PRIME, RECIP, S_BOX_U16, S_I};
 use crate::utils::divide_long_using_recip;
-use dusk_plonk::prelude::BlsScalar as Scalar;
-
-const DECOMPOSITION_LEN: usize = S_I_BLS.len();
+use dusk_plonk::prelude::BlsScalar;
 
 /// Convert representation from tuple in (Z_{s_n} x ... x Z_{s_1}) to single
 /// scalar element in Montgomery form (out of circuit)
```
Comment (vlopes11) [inline src/hash/zelbet/bar.rs]: Maybe unrelated to this PR, but its unclear what `scalar element in Montgomery form (out of circuit)` means here. Maybe rephrase this documentation?

**Review by xevisalle — APPROVED** [review]

File: src/constants.rs (hunk 1 above)
Comment (moCello) [reply src/constants.rs]: I use for the creation of the `MDS_MATRIX` constant a bit later. But if you prefer, I can also use `BlsScalar::one()` in the matrix.

File: src/hash/zelbet/bar.rs (hunk 2 above)
Comment (moCello) [reply src/hash/zelbet/bar.rs]: Good catch, fixed it in the following commits

**Review by vlopes11 — APPROVED** [review]
LGTM

**Review by xevisalle — APPROVED** [review]
